resources/install.rb

```diff
@@ -0,0 +1,71 @@
+# resources/install.rb
+
+# Resource for InfluxDB install
+
+property :arch_type, kind_of: String, default: 'amd64'
+property :include_repository, kind_of: [TrueClass, FalseClass], default: true
+property :influxdb_key, kind_of: String, default: 'https://repos.influxdata.com/influxdb.key'
+property :install_type, kind_of: String, default: 'package'
+default_action :install
+
+action :install do
+  case install_type
+  when 'package'
+    if node.platform_family? 'rhel'
+      yum_repository 'influxdb' do
+        description 'InfluxDB Repository - RHEL \$releasever'
+        baseurl node['influxdb']['upstream_repository']
+        gpgkey influxdb_key
+        only_if { include_repository }
+      end
+    else
+      package 'apt-transport-https' do
+        action :install
```

Comment:
I went through a few Test Kitchen setups mostly testing 12.7.  The changes as is work and pass functional tests.  

Before we merge this, I'd like to see if we can resolve the machine logic by letting the inline chef resources take care of that.  There are others simplifications that are nice to have but can be swayed to let these be another PR.